internal/terraform/exec/exec.go

```diff
@@ -59,12 +51,36 @@ func NewExecutor(path string) *Executor {
 		timeout:  defaultExecTimeout,
 		execPath: path,
 		logger:   log.New(ioutil.Discard, "", 0),
-		cmdCtxFunc: func(ctx context.Context, path string, arg ...string) *exec.Cmd {
-			return exec.CommandContext(ctx, path, arg...)
-		},
 	}
 }
 
+// tfExec should be used to initialize and return an instance of tfexec.Terraform just in
+// time, the instance will be cached. This is helpful because the language server sets the working
+// directory after the creation of the Executor instance.
+func (e *Executor) tfExec() *tfexec.Terraform {
+	if e.tf == nil {
+		tf, err := tfexec.NewTerraform(e.workDir, e.execPath)
+		if err != nil {
+			panic(err)
+		}
+		tf.SetLogger(e.logger)
+
+		// TODO: make sense of how this works
+		// if e.execLogPath != "" {
+		// 	logPath, err := logging.ParseExecLogPath(cmd.Args, e.execLogPath)
+		// 	tf.SetLogPath(logPath)
+		// }
+
+		// TODO: figure out what env vars are already handled by tfexec
```

Comment:
Still need to get env vars to whatever they were before, goal of the PR was to keep the "experience" the same as before (given my limited knowledge of the LS codebase at this time)